File: app/src/main/java/com/gritacademy/se/HistoryFragment.java
package com.gritacademy.se;

import android.os.Bundle;
import android.view.View;

import androidx.fragment.app.Fragment;

import android.widget.TextView;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.QueryDocumentSnapshot;

public class HistoryFragment extends Fragment {

    public HistoryFragment() {
        super(R.layout.fragment_history);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        TextView historyText = view.findViewById(R.id.historyText);

        FirebaseFirestore db = FirebaseFirestore.getInstance();

        db.collection("weatherHistory")
                .get()
                .addOnSuccessListener(documents -> {

                    String history = "";

                    for (QueryDocumentSnapshot document : documents) {
                        history += document.getString("city") + "\n";
                        history += document.getDouble("temperature") + " °C\n";
                        history += document.getDouble("wind") + " m/s\n";
                        //history += "---------------------";
                    }

                    historyText.setText(history);

                });




    }
}

File: app/src/main/java/com/gritacademy/se/HistoryFragment.kt
package com.gritacademy.se

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.google.android.gms.tasks.OnSuccessListener
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.QuerySnapshot

class HistoryFragment : Fragment(R.layout.fragment_history) {
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val historyText = view.findViewById<TextView>(R.id.historyText)

        //Firebase connection
        val db = FirebaseFirestore.getInstance()

        //Collects Firebase data and displays it in TextView
        db.collection("weatherHistory")
            .get()
            .addOnSuccessListener(OnSuccessListener { documents: QuerySnapshot? ->
                var history = ""
                for (document in documents!!) {
                    history += document.getString("city") + "\n"
                    history += document.getDouble("temperature").toString() + " °C\n"
                    history += document.getDouble("wind").toString() + " m/s\n"
                    //history += "---------------------";
                }
                historyText.setText(history)
            })
    }
}